client/components/client-track/clientTrack.js

```diff
@@ -0,0 +1,138 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import './client-track.css';
+import { useRouter } from 'next/navigation';
+import { getAppointment } from '../../actions/client-track';
+import { useClientContext } from '../../context/clientContext';
+
+//Form validation using yup
+import * as yup from 'yup';
```

Comment:
Not a big deal, but we have already added Joi for validation. If yup is preferred we can easily modify, but we should probably use one validation pkg.